react.gradle

```diff
@@ -116,6 +116,10 @@ afterEvaluate {
 
         def enableHermes = enableHermesForVariant(variant)
 
+        // Set up dev mode
+        def devEnabled = !(config."devDisabledIn${targetName}"
```

Comment:
What do you think about adding a new config value what'd work similarly to `devDisabledIn...` and `bundleIn...` specific to hermes release mode? Overloading `devDisabledIn` might not be the expected behavior for everyone.

Reply:
You're probably right, it will be unexpected behaviour for those who know about the config flag. The `devDisabledIn..` and `bundleIn..` are specific to the `react-native bundle` task as it stands today. 

I find the description in a vanilla `build.gradle` to be a bit more general though: 
```gradle
 *   // whether to disable dev mode in custom build variants (by default only disabled in release)
 *   // for example: to disable dev mode in the staging build type (if configured)
 *   devDisabledInStaging: true
```

And I wonder how many would disable `dev`-mode bundling in a non-`Release` build? It seemed appropriate to me to "merge" behaviour here.

There are a lot of other specific config flags in `build.gradle` though, so I'm definitely open for adding another config flag for "hermes release mode" if that seems appropriate.